ntrans_gui.py

```diff
@@ -131,6 +138,23 @@ def __init__(self) -> None:
             about_help_buttonframe, command=self.open_help_page, text="Help"
         )
 
+        about_button.pack(side="left")
+        help_button.pack(side="right")
+
+        # Progress Indication
+        self.progress_frame = ttk.Frame(mainframe)
+        self.progress_indicator = ProgressIndicator(self.progress_frame)
+
+        self.progress_indicator.progress_bar.grid(
+            column=0, row=0, padx=(0, 0), pady=(0, 0)
+        )
+        self.progress_indicator.percentage_label.grid(
+            column=1, row=0, padx=(5, 0), pady=(0, 0)
+        )
+        self.progress_indicator.cancel_button.grid(
+            column=0, row=1, columnspan=2, padx=(0, 0), pady=(0, 0)
+        )
+
```

Comment:
I would put this code inside `class ProgressIndicator`, so that it decides how it grids its own widgets.